qutip/solver/mcsolve.py

```diff
@@ -167,36 +167,30 @@ def mcsolve(H, state, tlist, c_ops=(), e_ops=None, ntraj=500, *,
     return result
 
 
-class _MCSystem(_MTSystem):
+class _MCRHS(_MultiTrajRHS):
     """
     Container for the operators of the solver.
     """
 
-    def __init__(self, rhs, c_ops, n_ops):
-        self.rhs = rhs
+    def __init__(self, H, c_ops, n_ops):
+        self.rhs = H
+
+        self.H = H
```

Comment:
Why both?
Here `H` is wrong here since it can be both an effective Hamiltonian or Liouvillian.

Reply:
You are right, changed it back